src/Stryker.Core/Stryker.Core/MutationTest/MutationTestProcess.cs

```diff
@@ -106,20 +124,29 @@ public void Mutate()
 
         public StrykerRunResult Test(StrykerOptions options)
         {
-            var logicalProcessorCount = Environment.ProcessorCount;                  
+            var logicalProcessorCount = Environment.ProcessorCount;
             var usableProcessorCount = Math.Max(logicalProcessorCount / 2, 1);
 
             if (options.MaxConcurrentTestrunners <= logicalProcessorCount)
             {
                 usableProcessorCount = options.MaxConcurrentTestrunners;
             }
 
-            Parallel.ForEach(_input.ProjectInfo.ProjectContents.Mutants.Where(x => x.ResultStatus == MutantStatus.NotRun),
+            var mutantsNotRun = _input.ProjectInfo.ProjectContents.Mutants.Where(x => x.ResultStatus == MutantStatus.NotRun).ToList();
+
+            _progressReporter.ReportInitialState(mutantsNotRun.Count());
```

Comment:
I think the reporter should keep track of this count. It knows the total amout of mutations and gets notified on every mutant tested. 

We could add a ` _reporter.OnStartMutantTestRun(mutant);` to keep track of the number of not runned mutants, running mutants and runned mutants.